infra/gravity/testcontext.go

```diff
@@ -72,23 +81,59 @@ func (c *TestContext) Error() error {
 	return c.err
 }
 
-// Checkpoint marks milestone within a test
+// WithFields assigns additional logging fields to this context
+func (c *TestContext) WithFields(fields logrus.Fields) *TestContext {
```

Comment:
I wanted to this to be able to add custom logging attributes (i.e. specify the node tag the test is using as it is different from the cluster name to due how GCE resources are structured.
There's no structural hierarchy as on Azure where resources could be grouped under `Resource Group`s - instead, everything is flat. Additionally, resource names are subject to RFC1035 which limits the length to 63 characters.